contracts/ColonyTask.sol

```diff
@@ -288,18 +264,39 @@ contract ColonyTask is ColonyStorage, DSMath {
   function finalizeTask(uint256 _id) public
   auth
   taskExists(_id)
-  taskWorkRatingsAssigned(_id)
+  taskWorkRatingsClosed(_id)
   taskNotFinalized(_id)
   {
     Task storage task = tasks[_id];
     IColonyNetwork colonyNetworkContract = IColonyNetwork(colonyNetworkAddress);
 
+    int8[3] memory ratingMultiplicator = [-10, 10, 15];
+
     for (uint8 roleId = 0; roleId <= 2; roleId++) {
       uint payout = task.payouts[roleId][token];
       Role storage role = task.roles[roleId];
-      uint8 rating = (roleId == EVALUATOR) ? 50 : role.rating;
-      int divider = (roleId == WORKER) ? 30 : 50;
-      int reputation = SafeMath.mulInt(int(payout), (int(rating)*2 - 50)) / divider;
+
+      uint8 rating = (roleId == EVALUATOR) ? 3 : role.rating;
+      int reputation = SafeMath.mulInt(int(payout), (int(ratingMultiplicator[rating - 1]))) / 10;
```

Comment:
In cases where rating is `0` , reputation will be `0` here right? Shouldn't the correct missing ratings be set before we calculate reputation?

Reply:
Yes, thanks, I just need to move line 280 below the reputation penalty I think. Will change it soon.